src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +80,154 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
     }
 }
 
-const writeDbHandler = async (request: ClaimRequest, claimedByUser: GuildMember): Promise<void> => {
+const writeDbHandler = async (request: ClaimRequest, dbBountyResult: BountyCollection, claimedByUser: GuildMember): Promise<BountyCollection> => {
     const db: Db = await MongoDbUtils.connect('bountyboard');
-	const bountyCollection = db.collection('bounties');
+    const bountyCollection = db.collection('bounties');
+    let claimedBounty: BountyCollection;
+    const currentDate = (new Date()).toISOString();
 
-	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
-		_id: new mongo.ObjectId(request.bountyId),
-		status: BountyStatus.open,
-	});
+    // If claiming an evergreen bounty, create a copy and use that
+    if (dbBountyResult.evergreen) {
+        const bountyRec: BountyCollection = JSON.parse(JSON.stringify(dbBountyResult));
```

Comment:
Now that `dbBountyResult` is of type `BountyCollection`, we no longer need these JSON calls

Reply:
Removed